File: rocket-bi-web/di-core/data-ingestion/domain/response/GoogleServiceAccountSource.ts
import { DataSourceType, DataSources } from '@core/data-ingestion';
import { DataSource } from '@core/data-ingestion/domain/response/DataSource';
import { SourceId } from '@core/common/domain';

export class GoogleServiceAccountSource implements DataSource {
  id: SourceId;
  orgId: string;
  displayName: string;
  credential: string;
  readonly className = DataSources.GoogleServiceAccountSource;
  lastModify: number;

  constructor(id: SourceId, orgId: string, displayName: string, credential: string, lastModify: number) {
    this.id = id;
    this.orgId = orgId;
    this.displayName = displayName;
    this.credential = credential;
    this.lastModify = lastModify;
  }

  static fromObject(obj: any): GoogleServiceAccountSource {
    return new GoogleServiceAccountSource(obj.id, obj.orgId, obj.displayName, obj.credential, obj.lastModify);
  }
}

File: rocket-bi-server/src/main/scala/co/datainsider/jobworker/domain/source/GoogleServiceAccountSource.scala
package co.datainsider.jobworker.domain.source

import co.datainsider.jobworker.domain.{DataSource, DataSourceType}
import co.datainsider.jobworker.domain.DataSourceType.DataSourceType
import co.datainsider.jobworker.domain.Ids.SourceId

case class GoogleServiceAccountSource(
    orgId: Long = -1,
    id: SourceId,
    displayName: String,
    credential: String
) extends DataSource {
  override def getId: SourceId = id

  override def getName: String = displayName

  override def getType: DataSourceType = DataSourceType.GoogleTokenCredential

  override def getConfig: Map[String, Any] =
    Map(
      "credential" -> credential
    )
}
